pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -287,6 +287,9 @@ func (r *Builder) mapMacStaticIps(vm *model.VM) (ipMap string, err error) {
 				if ipStack.Device != guestNetwork.Device {
 					continue
 				}
+				if ipStack.Network != "0.0.0.0" {
```

Comment:
Hmm what if there is a device which does not have this network?

Reply:
we can do `ipStack.Network != "0.0.0.0" and ipStack.Network != ""` ?

AFAIU the bug is that we should never use as gateway a stack that has network that is not "0.0.0.0"
In the clusters I checked we always have some network, usually "0.0.0.0" and in some case other ips ... that will introduce the bug.

In https://issues.redhat.com/browse/MTV-2408 it was suggested this is the correct behavior, and it matches the clusters I have access to. 
Do we have someone to ask for second opinion about this  ?

Reply:
@mnecas hi, I asked Nijin Ashok in https://issues.redhat.com/browse/MTV-2408

> Yaacov Zamir , I checked in multiple lab environments (both vSphere 8.x and vSphere 7.x) and the default route/gateway always have "0.0.0.0" as "network" for both Win and Linux VMs. IIUC, our intention is to pass default gateway to the v2v and filtering network="0.0.0.0" will help us to find the default gateway of the source VM, because other static routes will have their respective destination networks set on the 'network'.

Reply:
sounds good to me